contracts/contracts/strategies/GeneralizedUniswapV3Strategy.sol

```diff
@@ -0,0 +1,850 @@
+        // compute it every time?
+        uint160 sqrtRatioAX96;
+        uint160 sqrtRatioBX96;
+    }
+
+    // A lookup table to find token IDs of position using f(lowerTick, upperTick)
+    mapping(int48 => uint256) internal ticksToTokenId;
+    // Maps tokenIDs to their Position object
+    mapping(uint256 => Position) public tokenIdToPosition;
+    // Token ID of the position that's being used to provide LP at the time
+    uint256 public currentPositionTokenId;
+
+    // A deployed contract that's used to call methods of Uniswap V3's libraries despite version mismatch
+    IUniswapV3Helper internal uniswapV3Helper;
+
+    // Future-proofing
+    uint256[50] private __gap;
+
+    /**
+     * @dev Ensures that the caller is Governor or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            msg.sender == IVault(vaultAddress).strategistAddr() ||
+                msg.sender == governor(),
+            "Caller is not the Operator, Strategist or Governor"
+        );
+        _;
+    }
+
+    /**
+     * @dev Ensures that the caller is Governor, Strategist or Operator.
+     */
+    modifier onlyGovernorOrStrategistOrOperator() {
+        require(
+            msg.sender == operatorAddr ||
+                msg.sender == IVault(vaultAddress).strategistAddr() ||
+                msg.sender == governor(),
+            "Caller is not the Operator, Strategist or Governor"
+        );
+        _;
+    }
+
+    /**
+     * @dev Initialize the contract
+     * @param _vaultAddress OUSD Vault
+     * @param _poolAddress Uniswap V3 Pool
+     * @param _nonfungiblePositionManager Uniswap V3's Position Manager
+     * @param _token0ReserveStrategy Reserve Strategy for token0
+     * @param _token1ReserveStrategy Reserve Strategy for token1
+     * @param _operator Address that can manage LP positions on the V3 pool
+     * @param _uniswapV3Helper Deployed UniswapV3Helper contract
+     */
+    function initialize(
+        address _vaultAddress,
+        address _poolAddress,
+        address _nonfungiblePositionManager,
+        address _token0ReserveStrategy,
+        address _token1ReserveStrategy,
+        address _operator,
+        address _uniswapV3Helper
+    ) external onlyGovernor initializer {
+        positionManager = INonfungiblePositionManager(
+            _nonfungiblePositionManager
+        );
+        IUniswapV3Pool pool = IUniswapV3Pool(_poolAddress);
+        uniswapV3Helper = IUniswapV3Helper(_uniswapV3Helper);
+
+        token0 = pool.token0();
+        token1 = pool.token1();
+        poolFee = pool.fee();
+
+        address[] memory _assets = new address[](2);
+        _assets[0] = token0;
+        _assets[1] = token1;
+
+        super._initialize(
+            _poolAddress,
+            _vaultAddress,
+            new address[](0), // No Reward tokens
+            _assets, // Asset addresses
+            _assets // Platform token addresses
```

Comment:
`pTokens` aren't needed for this strategy, yeah. `InitializableAbstractStrategy.initialize()` would throw if I pass a empty array here. Couldn't pass address(0) either since `_setPTokenAddress()` (called in `initialize()` method) would throw. 

I thought of copying relevant code from `InitializableAbstractStrategy` and getting rid of `pToken` and `rewardToken` logics but that'd involve a lot of code duplication and might be harder to maintain in the long run. So, as of now, the Uniswap strategy just ignores the `pToken` and `rewardTokenAddresses` fields